apps/token-manager/contracts/TokenManager.sol

```diff
@@ -231,46 +235,38 @@ contract TokenManager is ITokenController, IForwarder, AragonApp {
     }
 
     // ITokenController fns
+    // `onTransfer()`, `onApprove()`, and `proxyPayment()` are callbacks from the MiniMe token
+    // contract and are only meant to be called through the managed MiniMe token that gets assigned
+    // during initialization.
 
     /*
-    * @dev Notifies the controller about a token transfer allowing the controller to decide whether to allow it or react if desired (only callable from the token)
+    * @dev Notifies the controller about a token transfer allowing the controller to decide whether
+    *      to allow it or react if desired (only callable from the token).
+    *      Initialization check is implicitly provided by `onlyToken()`.
     * @param _from The origin of the transfer
     * @param _to The destination of the transfer
     * @param _amount The amount of the transfer
     * @return False if the controller does not authorize the transfer
     */
-    function onTransfer(address _from, address _to, uint _amount) public isInitialized returns (bool) {
-        require(msg.sender == address(token), ERROR_ON_TRANSFER_WRONG_SENDER);
-
-        bool includesTokenManager = _from == address(this) || _to == address(this);
-
-        if (!includesTokenManager) {
-            bool toCanReceive = _isBalanceIncreaseAllowed(_to, _amount);
-            if (!toCanReceive || transferableBalance(_from, now) < _amount) {
-                return false;
-            }
-        }
-
-        return true;
+    function onTransfer(address _from, address _to, uint256 _amount) public onlyToken returns (bool) {
+        return _isBalanceIncreaseAllowed(_to, _amount) && _transferableBalance(_from, getTimestamp()) >= _amount;
     }
 
     /**
     * @dev Notifies the controller about an approval allowing the controller to react if desired
+    *      Initialization check is implicitly provided by `onlyToken()`.
     * @return False if the controller does not authorize the approval
     */
-    function onApprove(address, address, uint) public returns (bool) {
+    function onApprove(address, address, uint) public onlyToken returns (bool) {
         return true;
     }
 
     /**
-    * @notice Called when ether is sent to the MiniMe Token contract
+    * @dev Called when ether is sent to the MiniMe Token contract
+    *      Initialization check is implicitly provided by `onlyToken()`.
     * @return True if the ether is accepted, false for it to throw
     */
-    function proxyPayment(address) public payable returns (bool) {
-        // Sender check is required to avoid anyone sending ETH to the Token Manager through this method
-        // Even though it is tested, solidity-coverage doesnt get it because
-        // MiniMeToken is not instrumented and entire tx is reverted
-        require(msg.sender == address(token), ERROR_PROXY_PAYMENT_WRONG_SENDER);
+    function proxyPayment(address) public payable onlyToken returns (bool) {
```

Comment:
consider throwing an exception in proxyPayment already instead of relying on the caller (minime-token) to throw if the method returns false (unless it is a defined interfaces intended to be used that way)

Reply:
Not quite sure if I understood the exception:

> unless it is a defined interfaces intended to be used that way

We _are_ relying on the MiniMe token to revert on this false return, but this can only be called from the token (and should only be when it receives ETH).

Throwing an error ourselves seems a bit redundant, and we more or less completely trust the MiniMe implementation in the Token Manager.

Reply:
👍 looks good.